src/mock/filter.js

```diff
@@ -0,0 +1,62 @@
+import {checkDateIsToday} from "../util";
+
+const filterNames = [
+  `all`, `overdue`, `today`, `favorites`, `repeating`, `archive`
+];
+
+const countOverdueTasks = (tasks) => {
+  return tasks.reduce((accumulator, it) => accumulator + (it.dueDate ? 1 : 0), 0);
+};
+
+const countTodayTasks = (tasks) => {
+  return tasks.reduce((accumulator, it) => accumulator + (checkDateIsToday(it.dueDate) ? 1 : 0), 0);
+};
+
+const countFavoriteTasks = (tasks) => {
+  return tasks.reduce((accumulator, it) => accumulator + (it.isFavorite ? 1 : 0), 0);
+};
+
+const countRepeatingTasks = (tasks) => {
+  return tasks.reduce((accumulator, it) => accumulator + (Object.values(it.repeatingDays).some(Boolean) ? 1 : 0), 0);
+};
+
+const countArchiveTasks = (tasks) => {
+  return tasks.reduce((accumulator, it) => accumulator + (it.isArchive ? 1 : 0), 0);
+};
+
+const countTasks = (tasks, filterName) => {
+  let count = 0;
+  switch (true) {
+    case filterName === `all`:
+      count = tasks.length;
+      break;
+    case filterName === `overdue`:
+      count = countOverdueTasks(tasks);
+      break;
+    case filterName === `today`:
+      count = countTodayTasks(tasks);
+      break;
+    case filterName === `favorites`:
+      count = countFavoriteTasks(tasks);
+      break;
+    case filterName === `repeating`:
+      count = countRepeatingTasks(tasks);
+      break;
+    case filterName === `archive`:
+      count = countArchiveTasks(tasks);
+      break;
+  }
```

Comment:
Откуда такой необычный подход ?
```suggestion
  switch (filterName) {
    case `all`:
      count = tasks.length;
      break;
    case  `overdue`:
      count = countOverdueTasks(tasks);
      break;
    case  `today`:
      count = countTodayTasks(tasks);
      break;
    case  `favorites`:
      count = countFavoriteTasks(tasks);
      break;
    case  `repeating`:
      count = countRepeatingTasks(tasks);
      break;
    case  `archive`:
      count = countArchiveTasks(tasks);
      break;
  }
```